### ISSUE TITLE
Allow specifying DLLs as delay-loaded DLLs

### ISSUE THREAD

**Issue by yodantauber** [coapp/coapp.powershell#26]
When including DLLs in a package (e.g. using "bin: "), it is useful to be able to specify that these DLLs need to be delay-loaded. This is done by adding the DLL file name (without path) to the DelayLoadDLLs linker option.

Some important notes:
- It should be subject to conditions, so that the consumer of the package can be given the choice of whether the DLL is delay loaded or not.
- This should probably go to the redist & symbol packages, not the main package (?).
- Right now this can be done by adding `ItemDefinitionGroup.Link.DelayLoadDLLs += "dllname.dll"` to the targets section, under some custom condition. There is room for improvement: it would be better if the DLL file name did not have to be typed in again (too error prone) - I mean, I would expect this to be part of what "bin: " (or an alternative file list) does, when the consumer enables delay loading.
- Probably related feature (may be needed to implement this): a command like "add-each-file" that will ignore paths and add file names to the list.

**Comment by fearthecowboy** [comment]
Good idea. 

I feel like doing some coding today; I can quickly add this into my current build. I'll publish it today as the 'beta' version and you can start playing with it.

**Comment by yodantauber** [comment]
Great, thanks!

**Comment by yodantauber** [comment]
Quick tip:
I've found out (the hard way) that you may have to add delayimp.lib to the AdditionalDependencies explicitly if you specify DelayLoadDLLs in an imported targets file (which wouldn't be necessary if you specified DelayLoadDLLs in the vcxproj itself, as the linker would just do it implicitly).

**Comment by fearthecowboy** [comment]
Very good to know -- that's certainly a weird thing.

I wonder if we specified the DelayLoadDLLs in the props section (which gets evaluated early in the build) would fix that. Could you do a quick hack and try that?

FYI, I've pretty much finished the code reorg for the tools, I've put everything (including the clrplus stuff) into a single solution, and I should be pushing that up today.

**Comment by yodantauber** [comment]
You are correct. If I replace "targets" by "props" it works without explicitly specifying "delayimp.lib" as a dependency.
